### PR TITLE
Fix actions on layers

### LINKED ISSUE #417: [BUG] @JLayer changes Action behaviour.
**Describe the bug**

@JLayers changes an Action behaviour it shouldn't.

**To Reproduce**

1. Julia Version (i.e. output of `julia -v`): 1.6.2

2. Operating system (Mac, Linux, Windows): Windows 10

3. Javis version (i.e output of `] status Javis` in the REPL) 0.6.2

4. Minimum working code example that led to bug:

```julia
using Javis

nolayer_video = Video(500, 500)

function ground(args...)
    background("black") # canvas background
    sethue("white") # pen color
end

Background(1:140, ground)

circ = Object(JCircle(O, 20, action=:fill, color="white"))

act!(circ, Action(1:70, anim_translate(Point(150, 150))))
act!(circ, Action(70:140, anim_translate(Point(-150, -150))))

render(nolayer_video, pathname="mwe1.gif")


layer_video = Video(500, 500)

Background(1:140, ground)

l1 = @JLayer 1:140 begin
    Object(JCircle(O, 20, action=:fill, color="white"))
end

act!(l1, Action(1:70, anim_translate(Point(150, 150))))
act!(l1, Action(70:140, anim_translate(Point(-150, -150))))

render(layer_video, pathname="mwe2.gif")
```

mwe1.gif:
mwe1

mwe2.gif:
mwe2

### PR DESCRIPTION
**Pull request by gpucce**
## PR Checklist

If you are contributing to `Javis.jl`, please make sure you are able to check off each item on this list:

- [x] Did I update `CHANGELOG.md` with whatever changes/features I added with this PR?
- [x] Did I make sure to only change the part of the file where I introduced a new change/feature?
- [ ] Did I cover all corner cases to be close to 100% test coverage (if applicable)?
- [ ] Did I properly add Javis dependencies to the `Project.toml` + set an upper bound of the dependency (if applicable)?
- [ ] Did I properly add test dependencies to the `test` directory (if applicable)?
- [ ] Did I check relevant tutorials that may be affected by changes in this PR?
- [ ] Did I clearly articulate why this PR was made the way it was and how it was made?

**Link to relevant issue(s)**
Closes #417 

**How did you address these issues with this PR? What methods did you use?**
Now the action.func on layers doesn't set the position but adds to it and at the end of `get_javis_frame` I tried to reset the frames positions so that the action at the next frame add to the original position rather than the modified one. Added test derived from #417 where I check that the gif's frames with and without layer are the same.

### CODE REVIEW

File: src/Javis.jl [hunk 1]
```diff
@@ -526,9 +526,10 @@ function get_javis_frame(video, objects, frame; layers = Layer[])
 
     # check if any layers have been defined
     if !isempty(layers)
-
+        starting_positions = []
```
Comment (Wikunia) [inline src/Javis.jl:529]: Probably `Point[]` would be better here to have the type.

File: test/animations.jl [hunk 2]
```diff
@@ -715,3 +715,28 @@ end
...
+    
+    layer_video = Video(500, 500)
+    Background(1:n_frames, ground)
+    l1 = @JLayer 1:n_frames begin
+        Object(JCircle(O, 20, action=:fill, color="white"))
+    end
+    act_all!(l1)
+    render(layer_video, pathname="mwe2.gif", liveview=true)
+    
+    @test all(layer_video.layers[1].frames.frames .== nolayer_video.background_frames)
```
Comment (Wikunia) [inline test/animations.jl:741]: Here we would need test to compare that the gifs are actually the same.

File: test/animations.jl [hunk 3]
```diff
@@ -715,3 +715,52 @@ end
...
+
+    layer_video = Video(500, 500)
+    Background(1:n_frames, ground)
+    l1 = @JLayer 1:n_frames begin
+        Object(JCircle(O, 20, action = :fill, color = "white"))
+    end
+    act_all!(l1)
+    render(layer_video, tempdirectory = "images/with_layer", pathname = "")
+
+    @test_reference "images/with_layer/0000000008.png" load(
```
Comment (Wikunia) [inline test/animations.jl:740]: do you mind looping over them instead? :wink:

File: src/Javis.jl [hunk 4]
```diff
@@ -550,6 +551,9 @@ function get_javis_frame(video, objects, frame; layers = Layer[])
         end
 
         img_layers = place_layers(video, layers, frame)
+        for (position, layer) in zip(starting_positions, layers)
```
Comment (Wikunia) [inline src/Javis.jl:554]: Maybe just a comment 
`# resetting layer positions to originals` or something like that

**Review by Wikunia — COMMENTED** [review]
LGTM. I think a test case in which two actions are applied at the same time would be good too

File: test/animations.jl (hunk 3 above)
Comment (gpucce) [reply test/animations.jl:740]: sorry I don't understand what I should loop the tests? I add now the actions at the same time though it look like if I try that with anim_rotate_around it doesn't work, with anim_rotate it does

Comment (Wikunia) [reply test/animations.jl:740]: Just 
```julia
for frame in [8, 16, 32, 64, 128, 256]
    png_name = lpad(string(frame), 10, "0")
    @test_reference "images/with_layer/$png_name.png" load(
        "images/without_layer/$png_name.png",
    )
end
```
That should work I hope

Comment (gpucce) [reply test/animations.jl:740]: sure will do didn't understand

**Comment by Wikunia** [comment]
Great I think now it only needs a test case where `anim_translate` is applied in the same time frames like moving diagonally by moving it upward in one action and right in another action

**Comment by gpucce** [comment]
> Great I think now it only needs a test case where `anim_translate` is applied in the same time frames like moving diagonally by moving it upward in one action and right in another action

aaaah I added move and rotate do I add also the double translate?

**Comment by gpucce** [comment]
> > Great I think now it only needs a test case where `anim_translate` is applied in the same time frames like moving diagonally by moving it upward in one action and right in another action
> 
> aaaah I added move and rotate do I add also the double translate?

doing it now should be fast

**Comment by Wikunia** [comment]
Yeah we want to check that the adding works with the position and the rotate doesn't change that or does it? :thinking:

**Comment by gpucce** [comment]
> Yeah we want to check that the adding works with the position and the rotate doesn't change that or does it? 🤔

it makes perfect sense didn't think about it at first

**Comment by Wikunia** [comment]
Awesome one very last thing: One bugfix line in the changelog would be good

**Comment by Wikunia** [comment]
Thanks and sorry I have no idea what I did with git and the master branch... Somehow your tutorial is now in this branch which is really weird.

**Comment by gpucce** [comment]
I think I added one useless commit which I then tried to revert might be I had carried more stuff but didn't realize

**Comment by Wikunia** [comment]
I'm pretty sure it's my weird force push thing. I changed master back now but somehow it is still in this PR. However merging will probably fix this. I'll just let the test cases run again and merge.

**Comment by gpucce** [comment]
Great! Thanks for the step by step supervision!!

**Comment by Wikunia** [comment]
You're very welcome. Thanks for the fast fix :)
